src/shared/utils/parseDate.ts

```diff
@@ -0,0 +1,50 @@
+const edgeCase1 = /'\w{3}-\d{4}'/;
+
+const months = [
+  'jan',
+  'feb',
+  'mar',
+  'apr',
+  'may',
+  'jun',
+  'jul',
+  'aug',
+  'sep',
+  'oct',
+  'nov',
+  'dec',
+];
+
+/**
+ * Either returns a valid date, or undefined, but no invalid date
+ * @param input date to parse
+ * @returns {Date | undefined} valid date if possible, otherwise undefined
+ */
+const parseDate = (input?: string | number | Date | null): Date | undefined => {
+  if (!input) {
+    return undefined;
+  }
+  if (input instanceof Date) {
+    return input;
+  }
+  // easy case
+  let output = new Date(input);
+  if (!Number.isNaN(output.getTime())) {
+    return output;
+  }
+  /* edge cases */
+  // This one is specific to Firefox and Safari, as of April 2021.
+  // This will not be able to be covered by tests as node is using the V8 engine
+  if (typeof input === 'string' && edgeCase1.test(input)) {
```

Comment:
Could extract into a separate function (eg `parseDifficultDate`) to run unit tests in V8

Reply:
Good idea, I did that, and that even allowed me to catch an issue in my regular expression 👍🏽